File: app/src/main/java/org/ametro/ui/tasks/TaskHelpers.java
package org.ametro.ui.tasks;

import android.app.Activity;
import android.widget.Toast;

import org.ametro.R;

public class TaskHelpers {

    public static void displayFailReason(Activity activity, Throwable reason){
        if(reason instanceof InterruptedException){
            Toast.makeText(activity, activity.getString(R.string.msg_operation_interrupted), Toast.LENGTH_LONG).show();
        }else {
            Toast.makeText(activity, activity.getString(R.string.msg_error, reason.getMessage()), Toast.LENGTH_LONG).show();
        }
    }

}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/tasks/TaskHelpers.kt
package io.github.romangolovanov.apps.ametro.ui.tasks

import android.app.Activity
import android.widget.Toast
import io.github.romangolovanov.apps.ametro.R

object TaskHelpers {

    fun displayFailReason(activity: Activity, reason: Throwable) {
        if (reason is InterruptedException) {
            Toast.makeText(activity, activity.getString(R.string.msg_operation_interrupted), Toast.LENGTH_LONG).show()
        } else {
            Toast.makeText(activity, activity.getString(R.string.msg_error, reason.message), Toast.LENGTH_LONG).show()
        }
    }
}
